dri3_ws.c

```diff
@@ -399,22 +399,17 @@ static void handle_special_event(struct driws_drawable *drawable, xcb_present_ge
 
 		struct driws_buffer *buffer = find_buffer_from_list(ie->pixmap);
 
-		FAIL_IF(!buffer, "no buffer");
-
-		buffer->busy = false;
+		if (buffer)
+			buffer->busy = false;
 
 		break;
 	}
 
 	case XCB_PRESENT_EVENT_CONFIGURE_NOTIFY: {
+		__attribute__((unused))
 		xcb_present_configure_notify_event_t *ce = (xcb_present_configure_notify_event_t*) ge;
 		DBG("XCB_PRESENT_EVENT_CONFIGURE_NOTIFY %ux%u", ce->width, ce->height);
-
-		drawable->width = ce->width;
-		drawable->height = ce->height;
-
-		//create_buffers(drawable);
-
+		drawable->size_changed = true;
```

Comment:
size_changed is never set to false. Does size_changed = true always lead to destruction of the drawable?

Reply:
Yes, that is correct